src/components/ha-addon-picker.ts

```diff
@@ -0,0 +1,182 @@
+import "@polymer/paper-item/paper-item-body";
+import "@polymer/paper-dropdown-menu/paper-dropdown-menu-light";
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  internalProperty,
+  LitElement,
+  property,
+  PropertyValues,
+  query,
+  TemplateResult,
+} from "lit-element";
+import memoizeOne from "memoize-one";
+import { fireEvent } from "../common/dom/fire_event";
+import { compare } from "../common/string/compare";
+import { HassioAddonInfo } from "../data/hassio/addon";
+import { fetchHassioSupervisorInfo } from "../data/hassio/supervisor";
+import { HomeAssistant } from "../types";
+import { PolymerChangedEvent } from "../polymer-types";
+import { HaComboBox } from "./ha-combo-box";
+import { showAlertDialog } from "../dialogs/generic/show-dialog-box";
+import { isComponentLoaded } from "../common/config/is_component_loaded";
+
+const rowRenderer = (
+  root: HTMLElement,
+  _owner,
+  model: { item: HassioAddonInfo }
+) => {
+  if (!root.firstElementChild) {
+    root.innerHTML = `
+    <style>
+      paper-item {
+        margin: -10px 0;
+        padding: 0;
+      }
+    </style>
+    <paper-item>
+      <paper-item-body two-line="">
+        <div class='name'>[[item.name]]</div>
+        <div secondary>[[item.area]]</div>
+      </paper-item-body>
+    </paper-item>
+    `;
+  }
+
+  root.querySelector(".name")!.textContent = model.item.name;
+  root.querySelector("[secondary]")!.textContent = model.item.slug;
+};
+
+@customElement("ha-addon-picker")
+class HaAddonPicker extends LitElement {
+  public hass!: HomeAssistant;
+
+  @property() public label?: string;
+
+  @property() public value = "";
+
+  @internalProperty() private _addons?: HassioAddonInfo[];
+
+  @property({ type: Boolean }) public disabled = false;
+
+  @property({ type: Boolean }) private _opened?: boolean;
+
+  @query("ha-combo-box") private _comboBox!: HaComboBox;
+
+  private _init = false;
+
+  public open() {
+    this._comboBox?.open();
+  }
+
+  public focus() {
+    this._comboBox?.focus();
+  }
+
+  private _processedAddons = memoizeOne((addons?: HassioAddonInfo[]) => {
+    if (!addons) {
+      return [];
+    }
+    return addons.sort((a, b) => compare(a.name, b.name));
+  });
+
+  protected updated(changedProps: PropertyValues) {
+    if (
+      (!this._init && this._addons) ||
+      (changedProps.has("_opened") && this._opened)
+    ) {
+      this._init = true;
+      (this._comboBox as any).items = this._processedAddons(this._addons);
```

Comment:
Since add-ons never change (we fetch once) we don't have to keep track of open, just set items in render.